src/js/media-controller.js

```diff
@@ -981,6 +997,44 @@ const Delegates = {
 
     return null;
   },
+  [MediaUIAttributes.MEDIA_TIME_IS_LIVE](controller) {
+    const media = controller.media;
+    
+    if (!media) return false;
+
+    const streamIsLive = controller.getAttribute(MediaUIAttributes.MEDIA_STREAM_TYPE) === 'live';
+
+    // If there's no way to seek, assume the media element is keeping it "live"
+    if (streamIsLive && !media.seekable) {
+      return true;
+    }
+
+    if (media.seekable.length === 0) {
+      return false;
+    }
+
+    // Default to 10 seconds
+    // Assuming seekable range already accounts for appropriate buffer room
+    const liveTimeMargin = 10;
```

Comment:
idea(non-blocking): This is the sort of thing that could be more reliably computed by a(n extended) slotted media element based on spec-compliant/recommended behaviors on refetches, segment sizes, etc. etc. Would be a good followup to spike on.

Reply:
Yeah, I agree. That'd be a good way to set your latency preferences on the media element specifically and have it impact the right things in the UI, without latency needing to be a bigger concept (at least yet).

Start a media ui extension issue?

Reply:
@heff Yeah, good call re: media ui extension.